### PR TITLE
Humbug reporter for moonstream backend

### PR DESCRIPTION
**Pull request by kompotkot**
## Changes

Gumbug reporter for backend.

## How to test these changes?

Tested manually.

### CODE REVIEW

**Comment by kompotkot** [comment]
@bugout-dev check
- [ ] Add env var `HUMBUG_REPORTER_BACKEND_TOKEN`

File: backend/moonstream/routes/streams.py [hunk 1]
```diff
@@ -121,17 +123,27 @@ async def stream_handler(
...
+            MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            stream_boundary,
+            query,
+            user_subscriptions,
+            result_timeout=10.0,
+            raise_on_error=True,
+        )
+    except ReceivingEventsException as e:
+        logger.error("Error receiving events from provider")
+        reporter.error_report(e)
```
Comment (zomglings) [inline backend/moonstream/routes/streams.py:140]: From the API, can we report in `background_task`? It doesn't make sense to require the user to wait for our error reporting to complete before they get a response.

**Review by zomglings — CHANGES_REQUESTED** [review]
All reporting done in API endpoints should report errors in a background task.

One thing you can consider is adding a middleware to all API endpoints. If handler raises exception, report in background and reraise.

File: backend/moonstream/routes/streams.py (hunk 1 above)
Comment (zomglings) [reply backend/moonstream/routes/streams.py:140]: Actually, I think if you instantiate the reporter with default configuration, it sends reports in a thread.

Use [`mode=DEFAULT`](https://github.com/bugout-dev/humbug/blob/main/python/humbug/report.py#L60) and make sure `wait=False` on report calls. I think this is okay in your current code.

Comment (zomglings) [reply backend/moonstream/routes/streams.py:140]: Background tasks not needed, but I do recommend middleware.

Comment (kompotkot) [reply backend/moonstream/routes/streams.py]: Extended class `HTTPException` and `Response`

**Review by zomglings — APPROVED** [review]
lgtm

**Pull request merged**